mypy/checker.py

```diff
@@ -5285,7 +5339,7 @@ def visit_decorator_inner(self, e: Decorator, allow_empty: bool = False) -> None
         # For overloaded functions we already checked override for overload as a whole.
         if allow_empty:
             return
-        if e.func.info and not e.func.is_dynamic() and not e.is_overload:
```

Comment:
This is a major change, though. IMO the updated behavior is correct: even if the function is untyped, it still shouldn't be allowed to have invalid `@override` attached. We won't check its superclass compatibility unless `--check-untyped-defs` is set (`check_override_compatibility` flag down the road).